Update zds-vs-zgw.md (#984)
VNG-Realisatie/gemma-zaken: docs/_content/overige/technisch/zds-vs-zgw.md

APIs, maar er is ook overlap. De belangrijkste verschillen en overeenkomsten op
een rij:

[Zaak-Documentservices](https://www.gemmaonline.nl/index.php/Zaak-_en_Documentservices) | [ZGW APIs](https://ref.tst.vng.cloud/standaard/index)\*
--- | ---

> SOAP/XML gebaseerd	                                                                        | REST/JSON gebaseerd

Reviewer: Dit is dus de achterliggende techniek en het (standaard) formaat.

Author: begrijp ik goed dat je liever een aantal paragrafen met daaronder vergelijkingen tussen ZDS en ZGW API's op dat vlak ziet?

Reviewer: Ja, maar ik weet niet of het dus persé beter wordt, maar ik vind de opzet van de tabel niet duidelijk geworden.

Reviewer 2: Voor een goed vergelijk moet het verschil tussen de onderliggende architecturen van de beide standaarden uiteen gezet worden. ZDS veronderstelt synchroniseren, micro services gaat uit van gegevens bij de bron. Zoals Andy gister ook terecht opmerkte zijn de ZGW API's niet een directe vervanger van ZDS, maar veronderstelt een andere architectuur van applicaties en het gegevenslandschap. Daar moeten we iets over zeggen. Dat kan in dit document. Maar ik zal dat ook in de productvisie laten landen.

Gebaseerd op het informatiemodel RGBZ1	                                                    | Gebaseerd op informatiemodel RGBZ2 waar van toepassing
Verdere invulling/uitwerking van de basisberichten uit StUF ZKN	                            | Aparte calls voor aparte resources

> Gebaseerd op uitwisselen van informatie, gestructureerd volgens het informatiemodel [RGBZ 1](https://www.gemmaonline.nl/index.php/Informatiemodel_Zaken_(RGBZ))	| Gebaseerd op operaties op resources die leven in het referentiecomponent

Reviewer: Beetje scheve vergelijking?

Beide zijn gebaseerd op uitwisselen van informatie, maar.. het is StUF-ZKN vs OASen van de ZGW API's

Ook klinkt "...leven in het referentiecomponent" vreemd, want het referentiecomponent is alleen maar een hulpmiddel bij de standaard.

Author: Ik zie toch wel verschil: in StUF berichten kunnen gegevens opgenomen zijn die niet of niet volledig in het Zaaksysteem opgeslagen worden. Bovendien worden in StUF berichten alle soorten gegevens in 1 bericht (bv om een zaak aan te maken) gestopt. Terwijl bij de ZGW API's de resources apart opgevoerd worden en alleen een verwijzing naar de overige resources opgenomen wordt in de zaak.

Author: Leven in referentiecomponent moet eigenlijk zijn leven in een registratie.

Reviewer: > Ik zie toch wel verschil

Er is ook een verschil, maar ik vind het scheef lopen dat je aan de ene kant "structuur RGBZ" noemt, en aan de andere kant niets over structuur zegt.

Author: Wat ik wil zeggen is dat StUF ZKN en de daarvan afgeleide koppelvlakken de structuur van het RGBZ volgen. Bij het opzetten van de API's is deze, "verplichte", structuur losgelaten en gekeken welke resources en attributen van toepassing zijn op de user stories.
Uiteindelijk is er een model uitgekomen wat voor een heel groot deel overeenkomt met het RGBZ maar dat model is niet vantevoren opgelegd, zoals dat bij StUF ZKN wel zo is.
Hopelijk is een beetje duidelijk wat ik bedoel, ik ben het vehaal toch aan het omschrijven, hopelijk komt het dan beter uit de verf.

Reviewer 3: Ik vind de nieuwe formulering prima :+1:

> Gebaseerd op uitwisselen van informatie, gestructureerd volgens het informatiemodel [RGBZ 1](https://www.gemmaonline.nl/index.php/Informatiemodel_Zaken_(RGBZ))	| Gebaseerd op operaties op resources die leven in het referentiecomponent

Reviewer 3: ```suggestion
Uitwisselen van informatie inclusief gerelateerde gegevens, gestructureerd volgens het informatiemodel [RGBZ 1](https://www.gemmaonline.nl/index.php/Informatiemodel_Zaken_(RGBZ))	| Uitwisselen van informatie op basis van atomische resources, waarbij er gerefereerd wordt naar een resource in een component in plaats van deze te embedden.
```

Author: bedankt, goede suggestie, overgenomen.

Ondersteunen van de functies van een zaaksysteem zoals beschreven in [GEMMA 1](https://www.gemmaonline.nl/index.php/Overzicht_GEMMA_1_referentiecomponenten)	            | Ondersteunen van functies van de referentiecomponenten zoals beschreven in [GEMMA 2](https://www.gemmaonline.nl/index.php/Overzicht_generieke_referentiecomponenten_GEMMA_2)
Kent alleen de functionele acties die uitgewerkt zijn in de standaard.	                    | Alle functionele acties die mogelijk zijn met de CRUD functionaliteit zijn mogelijk. De ZGW API's zijn microservices die gecombineerd kunnen worden om een functionele behoefte in te vullen.

> Betrokkenen en/of gerelateerde objecten kunnen geheel of gedeeltelijk, in ieder geval met matchinggegevens doorgegeven worden. Deze worden veelal in het zaaksysteem opgeslagen.	                                                             | een relatie naar een andere resource is een verwijzing naar die resource. Die resource wordt opgeslagen in de betreffende registratie. Dat kan een zaakregistratie (ZRC) zijn maar ook een andere (met een eigen API). 

Reviewer: Dit is wel een goede om op te merken. Het kopje "Relaties naar andere Objecttypen/Resources" zou er zo boven kunnen als je dat documenteer formaat aanhoudt.

> Betrokkenen en/of gerelateerde objecten kunnen geheel of gedeeltelijk, in ieder geval met matchinggegevens doorgegeven worden. Deze worden veelal in het zaaksysteem opgeslagen.	                                                             | een relatie naar een andere resource is een verwijzing naar die resource. Die resource wordt opgeslagen in de betreffende registratie. Dat kan een zaakregistratie (ZRC) zijn maar ook een andere (met een eigen API). 

Reviewer 3: ```suggestion
Betrokkenen en/of gerelateerde objecten kunnen geheel of gedeeltelijk, in ieder geval met matchinggegevens doorgegeven worden. Deze worden veelal in het zaaksysteem opgeslagen.	                                                             | Een relatie naar een andere resource is een verwijzing naar die resource. Die resource wordt opgeslagen in de betreffende registratie. Dat kan een zaakregistratie (ZRC) zijn maar ook een andere (met een eigen API). 
```

> bestaan voornamelijk uit a-synchrone services	                                            | bestaan uit alleen maar synchrone services

Reviewer 3: ```suggestion
Bestaan voornamelijk uit a-synchrone services	                                            | Bestaan uit alleen maar synchrone services
```

Een [GEMMA 1](https://www.gemmaonline.nl/index.php/Overzicht_GEMMA_1_referentiecomponenten) Zaaksysteem bevat ook (generieke) zaakafhandelfunctionaliteit. Deze hoeft niet per se via ZDS te communiceren met het eigen zakenmagazijn  | Een [GEMMA 2](https://www.gemmaonline.nl/index.php/Overzicht_generieke_referentiecomponenten_GEMMA_2) [Zaakregistratiecomponent](https://www.gemmaonline.nl/index.php/GEMMA2/0.9/id-a97b6545-d5a7-485d-9b13-3ce22db5b9cf) registreert alleen zaken en (verwijzingen naar) gerelateerden. Afhandelen gebeurt in een [zaakafhandelcomponent](https://www.gemmaonline.nl/index.php/GEMMA2/0.9/id-f2dfbd0b-9d36-405c-bdbe-827f3296de29) (ZAC). Communicatie vindt plaats via de ZGW API's

> Zaak- Documentservices is gemaakt om taakspecifieke applicaties te koppelen aan een zaaksysteem. | ZGW API's zijn gemaakt om zaakgericht werken te ondersteunen

Reviewer: Jij weet dit beter dan ik maar is het speciaal gemaakt voor "taakspecifieke applicaties"? Het is toch vooral dat het een "eenvoudigere laag op StUF-ZKN" is?

Author: Eigenlijk heb ik het ook iets te kort door de bocht beschreven. Officieel is ZDS (par 1.3 UItgangspunten en reikwijdte, eerste bullet):
* De services ondersteunen de meest gebruikte basisfunctionaliteit voor de 2-weg applicatie-applicatiekoppeling tussen een DMS en ZS onderling en de koppelingen van een DMS en ZS met andere systemen voor het ontsluiten en muteren van zaakgegevens en zaakdocumenten;

Je kunt ZDS dus ook gebruiken voor bijvoorbeeld zaakbevragingen of om een zakenmagazijn over te zetten naar een ander zakenmagazijn. Iets wat ook kan met de ZGW API's.

Wat ik wilde aangeven is dat het hele proces van zaakgericht werken, dus inclusief het opvragen van zaaktypen etc. in de ZTC, uitgewerkt is in de ZGW API's. In ZDS is slechts het meest gebruikte deel uitgewerkt vanuit het gezichtspunt van de TSA.

Zaak- Documentservices beschrijven alleen hoe zaken, documenten en besluiten in een zaaksysteem vastgelegd kunnen worden. Er wordt niet beschreven hoe een ZTC bevraagd moet worden etc.                                                           | ZGW API's ondersteunen alle onderdelen van het zaakgericht werken, ook het bevragen van een ZTC etc.

> De Zaak- Documentservices standaard bestaat uit een specificatiedocument, xsd schema's en documentatie van de onderliggende StUF en CMIS standaarden.                                                                                | ZGW API standaard bestaat per API uit een OAS 3 yaml bestand, functionele en technische documentatie,  referentie implementaties

Reviewer 3: ```suggestion
De Zaak- Documentservices standaard bestaat uit een specificatiedocument, xsd schema's en documentatie van de onderliggende StUF en CMIS standaarden.                                                                                | ZGW API standaard bestaat per API uit een OpenAPI 3 specificatie (OAS), functionele en technische documentatie en  referentie-implementaties
```

> De Zaak- Documentservices standaard bestaat uit een specificatiedocument, xsd schema's en documentatie van de onderliggende StUF en CMIS standaarden.                                                                                | ZGW API standaard bestaat uit een OAS 3 yaml bestand, functionele en technische documentatie,  referentie implementaties (voorbeeld en proof of concept)

Reviewer: > ZGW API standaard bestaat uit [...]

per API :)

> (voorbeeld en proof of concept)

Graag weglaten. Dit staat ergens anders wel uitgelegd en ik wil dit niet overal wijzigen als de politiek weer een andere definitie wil gaan aanhouden :)

> Beheer van de Zaak- Documentservices gebeurt in de werkgroep Beheer Zaak- Documentservices. Daarnaast volgt de standaard de ontwikkelingen in de StUF standaard (voor zover van toepassing).                                                      | Beheer van de ZGW API standaard zal plaats vinden onder regie van [VNG-Realisatie](https://www.vngrealisatie.nl/). Belanghebbenden kunnen wensen en bevindingen of zelfs wijzigingsvoorstellen indienen via issues en pullrequests. 

Reviewer 3: ```suggestion
Beheer van de Zaak- Documentservices gebeurt in de werkgroep Beheer Zaak- Documentservices. Daarnaast volgt de standaard de ontwikkelingen in de StUF standaard (voor zover van toepassing).                                                      | Beheer van de ZGW API standaard zal (open source) plaatsvinden onder regie van [VNG-Realisatie](https://www.vngrealisatie.nl/). Belanghebbenden kunnen wensen, bevindingen of zelfs wijzigingsvoorstellen indienen via issues en pull requests. 
```

> Beheer van de Zaak- Documentservices gebeurt in de werkgroep Beheer Zaak- Documentservices. Daarnaast volgt de standaard de ontwikkelingen in de StUF standaard (voor zover van toepassing).                                                      | Beheer van de standaard zal plaats vinden onder regie van [VNG-Realisatie](https://www.vngrealisatie.nl/). Belanghebbenden kunnen wensen en bevindingen of zelfs wijzigingsvoorstellen indienen via issues en pullrequests. 

Reviewer: ```suggestion
Beheer van de Zaak- Documentservices gebeurt in de werkgroep Beheer Zaak- Documentservices. Daarnaast volgt de standaard de ontwikkelingen in de StUF standaard (voor zover van toepassing).                                                      | Beheer van de ZGW API standaard zal plaats vinden onder regie van [VNG-Realisatie](https://www.vngrealisatie.nl/). Belanghebbenden kunnen wensen en bevindingen of zelfs wijzigingsvoorstellen indienen via issues en pullrequests. 
```

Author: Met plezier neem ik je suggestie over maar ik zie het verschil even niet? :-D

> Zaak- Documentservices kent extraElementen, label-value combinaties voor gegevens die niet in het RGBZ 1 opgenomen zijn.  | ZGW API's werken op resources. Wanneer een attribuut of zelfs een resource niet bekend is moet deze toegevoegd worden aan de API die bij de juiste registratie hoort.

Reviewer: Dit geeft volgens mij een verkeerd beeld door het zinsdeel:

> Wanneer een attribuut of zelfs een resource **niet bekend is** [...]

Het gaat er niet om dat het niet bekend is, maar dat we niet allerlei domein data bij een zaak kunnen opslaan.

Als we het kopje "Additionele informatie opslaan bij een Zaak" nemen, zou ik eerder aangeven:

ZDS biedt de mogelijkheid om middels `extraElementen` informatie die niet in de standaard is opgenomen, toch toe te voegen aan bepaalde objecten, zoals een zaak. In de ZGW APIs is er voor gekozen deze constructie niet mogelijk te maken om compatibiliteit te bewaren. Wel kent een zaak bijvoorbeeld `eigenschappen` die in het ZTC zijn gedefinieerd op het zaaktype en in een zaak van een waarde kunnen worden voorzien. Als er zoveel extra informatie is dat `eigenschappen` niet werkbaar is, dan kan er altijd een nieuwe registratie API worden gerealiseerd en naar deze API worden verwezen middels een `zaakobject` (bijvoorbeeld een wegdeel), of vanuit de nieuwe API naar een zaak (bijvoorbeeld een melding).

Author: Domeindata zou nog kunnen wanneer dit uitgewerkt is. Het principe achter extraElementen en aanvullendeElementen is echter dat in een koppeling gegevens gebruikt kunnen worden die helemaal niet in het informatiemodel of de StUF berichten beschreven is. Bijvoorbeeld proprietary velden of een eigen enumeratie van een bestaand veld oid. 
Eigenschappen zijn netjes beschreven in de ZTC, dat is mooi maar dat zijn wel eigenschappen op zaaktype-niveau. Die gegevens zijn controleerbaar omdat ze uitgewerkt staan in de ZTC.
Een eigen key-value veld is echt maatwerk. Bovendien is er de spelregel dat wanneer een ontvanger het gegeven niet kent (de key komt niet in zijn lijstje voor) dat dit gegeven dan genegeerd kan worden. Dit maakt het makkelijker on the fly gegevens toe te voegen maar het wordt er niet overzichtelijker van.

Reviewer 2: Let op: inderdaad is er de mogelijkheid om domeinspecifieke gegevens als eigenschappen bij een zaaktype te definieren. Echter: vaak is de behoefte aan dit soort gegevens een aanwijziging voor een objecttype dat in een eigen registratie moet worden ondergebracht. In mijn visie moeten eigenschappen zo min mogelijk worden gebruikt en moet er worden gestreefd naar aparte registraties voor entiteiten buiten ZGW.
